Survival-DEV/blog-backend
# Blog-1: cleaning up the code

### :dart:  What is been done in this PR:
- [x] Install these packages
  - [x] [Helmet](https://docs.nestjs.com/security/helmet) ~ "^4.6.0"
  - [x] [Nest-morgan](https://github.com/mentos1386/nest-morgan) "^2.0.0" ships with "[morgan](https://github.com/expressjs/morgan)": ~ "^1.10.0"
  - [x] [Csurf](http://expressjs.com/en/resources/middleware/csurf.html) ~ "^1.11.0"
  - [x] [Cookieparser](https://docs.nestjs.com/techniques/cookies) ~ "^1.4.5"
  - [x] [Throttler](https://github.com/nestjs/throttler) ~ "^2.0.0" 
- [x] I have added `ThrottlerStorageRedisService` for Rate limiting middleware storage, but I removed it currently to re-make it in another dedicated issue for Redis only    
 
**Relates to #1**

----
### :compass:   How to test it
Steps to test my PR:
1. `npm run ci`
2. make sure to setup your `env` file to have these params:
> DATABASE_URL=<your db url#>
> NODE_ENV=DEBUG  //for now
> logDB=["query", "error"]
> PORT=<your port#>
> TRUSTED_IP=["127.0.0.1"]
> WHITELIST=['http://localhost']

6. `npm run start:dev`
